pages/activity/[id]/agenda/[agendaId]/invitation.tsx

```diff
@@ -0,0 +1,107 @@
+import { GetServerSideProps, InferGetServerSidePropsType } from 'next';
+
+import { FC, useEffect, useRef, useState } from 'react';
+
+import html2canvas from 'html2canvas';
+import { Loading } from 'idea-react';
+import { QRCodeSVG } from 'qrcode.react';
+import { Container, Image } from 'react-bootstrap';
+
+import { i18n } from '../../../../../models/Translation';
+import { Activity, ActivityModel } from '../../../../../models/Activity';
+import { Agenda } from '../../../../../models/Agenda';
+import { API_Host } from '../../../../../models/Base';
+import styles from './invitation.module.less';
+
+const { t } = i18n;
+
+export const getServerSideProps: GetServerSideProps<
+  { activity: Activity; agenda: Agenda; currentUrl: string },
+  { id: string; agendaId: string }
+> = async ({ resolvedUrl, params }) => {
+  const activityStore = new ActivityModel();
+  const { id, agendaId } = params!;
+  const activity = await activityStore.getOne(id + '');
+  const agenda = await activityStore.currentAgenda!.getOne(agendaId + '');
+  const currentUrl = API_Host + resolvedUrl;
+  return {
+    props: {
+      activity,
+      agenda,
+      currentUrl,
+    },
+  };
+};
+
+const Invitation: FC<
+  InferGetServerSidePropsType<typeof getServerSideProps>
+> = ({ activity, agenda, currentUrl }) => {
+  const elementRef = useRef<HTMLDivElement>(null);
+  const [imageDataURL, setImageDataURL] = useState('');
+
+  useEffect(() => {
+    generateImage();
+  }, []);
+
+  const { name, city, location } = activity;
+  const { startTime, endTime, title, mentors } = agenda;
+
+  const share = async () => {
+    try {
+      await navigator.share?.({
+        title: title as string,
+        text: '',
+        url: currentUrl,
+      });
+    } catch (error) {
+      console.log('share error', error);
+    }
+  };
+
+  const generateImage = async () => {
+    const element = elementRef.current;
+
+    try {
+      const canvas = await html2canvas(element!);
+      const image = canvas.toDataURL('image/jpeg', 0.92);
+      setImageDataURL(image as string);
```

Comment:
```suggestion
      const image = await new Promise<string>(resolve =>
        canvas.toBlob(blob => resolve(URL.createObjectURL(blob)), 'image/jpeg', 0.92)
      );
      setImageDataURL(image);
```
https://developer.mozilla.org/zh-CN/docs/Web/API/HTMLCanvasElement/toBlob